pandas/core/ops.py

```diff
@@ -990,6 +991,20 @@ def _construct_divmod_result(left, result, index, name, dtype):
     )
 
 
+def dispatch_to_extension_op(left, right, op_name):
```

Comment:
Could this be accomplished with:

```
return dispatch_to_index_op(op, left, right, left.values.__class__)
```
?

Reply:
Unless we require that the ExtensionArray constructor handles objects of itself (and does not copy them), we cannot do the your example code (and currently we don't require this from extension array implementations I think. 
It is also don't think it necessarily improves readability of the code by reusing that.

Reply:
Fair enough.  Eventually dispatch_to_index_op is intended to give way to an EA-based dispatch.